src/main/scala/firrtl/Emitter.scala

```diff
@@ -987,42 +1068,45 @@ class VerilogEmitter extends SeqTransform with Emitter {
     passes.VerilogPrep,
     new AddDescriptionNodes)
 
-  def emit(state: CircuitState, writer: Writer): Unit = {
-    val circuit = runTransforms(state).circuit
-    val moduleMap = circuit.modules.map(m => m.name -> m).toMap
-    circuit.modules.foreach {
+  def emit(state: CircuitState, writer: Writer): CircuitState = {
+    val cs = runTransforms(state)
+    val emissionOptions = new EmissionOptions(cs.annotations)
+    val moduleMap = cs.circuit.modules.map(m => m.name -> m).toMap
+    cs.circuit.modules.foreach {
       case dm @ DescribedMod(d, pds, m: Module) =>
-        val renderer = new VerilogRender(d, pds, m, moduleMap)(writer)
+        val renderer = new VerilogRender(d, pds, m, moduleMap, cs.circuit.main, emissionOptions)(writer)
         renderer.emit_verilog()
       case m: Module =>
-        val renderer = new VerilogRender(m, moduleMap)(writer)
+        val renderer = new VerilogRender(m, moduleMap, cs.circuit.main, emissionOptions)(writer)
         renderer.emit_verilog()
       case _ => // do nothing
     }
+    cs
   }
 
   override def execute(state: CircuitState): CircuitState = {
     val newAnnos = state.annotations.flatMap {
       case EmitCircuitAnnotation(_) =>
         val writer = new java.io.StringWriter
-        emit(state, writer)
-        Seq(EmittedVerilogCircuitAnnotation(EmittedVerilogCircuit(state.circuit.main, writer.toString, outputSuffix)))
+        val cs = emit(state, writer)
+        Seq(EmittedVerilogCircuitAnnotation(EmittedVerilogCircuit(state.circuit.main, writer.toString, outputSuffix))) ++ (cs.annotations.toSeq diff state.annotations.toSeq)
 
       case EmitAllModulesAnnotation(_) =>
-        val circuit = runTransforms(state).circuit
-        val moduleMap = circuit.modules.map(m => m.name -> m).toMap
+        val cs = runTransforms(state)
+        val emissionOptions = new EmissionOptions(cs.annotations)
+        val moduleMap = cs.circuit.modules.map(m => m.name -> m).toMap
 
-        circuit.modules flatMap {
+        cs.circuit.modules flatMap {
           case dm @ DescribedMod(d, pds, module: Module) =>
             val writer = new java.io.StringWriter
-            val renderer = new VerilogRender(d, pds, module, moduleMap)(writer)
+            val renderer = new VerilogRender(d, pds, module, moduleMap, cs.circuit.main, emissionOptions)(writer)
             renderer.emit_verilog()
-            Some(EmittedVerilogModuleAnnotation(EmittedVerilogModule(module.name, writer.toString, outputSuffix)))
+            Some(EmittedVerilogModuleAnnotation(EmittedVerilogModule(module.name, writer.toString, outputSuffix))) ++ (cs.annotations.toSeq diff state.annotations.toSeq)
```

Comment:
I don't understand what ` ++ (cs.annotations.toSeq diff state.annotations.toSeq)` is for. Similar to my previous point about not returning `cs` since we don't want the changes from the `VerilogEmitter` transforms to be exposed, shouldn't we do the same thing about the annotations?

Reply:
If I'm right, then we can restore `def emit` to returning `Unit`.

Reply:
you're right I guess this was only a debug feature for me that leaked into the PR
fixed in rebase